l5kit/l5kit/data/map_api.py

```diff
@@ -174,7 +177,8 @@ def get_crosswalk_coords(self, element_id: str) -> dict:
             traffic_element.geo_frame,
         )
 
-        return {"xyz": xyz}
+        xyz[:, -1] = 1.0
```

Comment:
why is this required? We ignore the z coordinates in the following so this shouldn't make any difference right?

Reply:
In the current implementation, first you cut the z-coordinate `lane_coords["xyz_right"][:, :2]` and later in the transform method you stack it back `np.vstack((points[:num_dims, :], np.ones(points.shape[1])))`. With this we can save the one cut, the np.ones creation, the stack.

Reply:
Another way of doing this would be to only use the first 2x2 of the matrix (only XY) in the semantic rasterizer. My issue with setting 1 here is that we're removing information in a very hidden part of the code, which may render debugging problematic in the future (also considering there is a cache system in between)

Reply:
I see. You can move it to the semantic rasterizer method; before the dot product calls.

Reply:
I like this idea also :) Today was quite busy, but I should be able to work on this tomorrow (hopefully)